contracts/javascore/xcall/src/test/java/foundation/icon/xcall/CallServiceTest.java

```diff
@@ -343,6 +365,37 @@ public void executeCall_failedExecution() {
         verify(xcallSpy).CallExecuted(BigInteger.ONE, 0, "score.RevertedException");
     }
 
+    @Test
+    public void executeCall_persistent_failedExecution() {
+        // Arrange
+        byte[] data = "test".getBytes();
+        CSMessageRequest request = new CSMessageRequest(ethDapp.toString(), dapp.getAddress().toString(), BigInteger.ONE, PersistentMessage.TYPE, data, baseSource);
+        CSMessage msg = new CSMessage(CSMessage.REQUEST, request.toBytes());
+        xcall.invoke(baseConnection.account, "handleMessage", ethNid, msg.toBytes());
+        // Act
+        doThrow(new UserRevertedException()).when(dapp.mock).handleCallMessage(ethDapp.toString(), data, new String[]{baseConnection.getAddress().toString()});
+        assertThrows(Exception.class, () -> xcall.invoke(user, "executeCall", BigInteger.ONE, data));
+    }
+
+    @Test
+    public void executeCall_persistent() throws Exception {
+        // Arrange
+        byte[] data = "test".getBytes();
+        MockContract<DefaultCallServiceReceiver> defaultDapp = new MockContract<>(DefaultCallServiceReceiverScoreInterface.class, DefaultCallServiceReceiver.class, sm, owner);
+        CSMessageRequest request = new CSMessageRequest(ethDapp.toString(), defaultDapp.getAddress().toString(), BigInteger.ONE, PersistentMessage.TYPE, data, null);
+        CSMessage msg = new CSMessage(CSMessage.REQUEST, request.toBytes());
+
+        xcall.invoke(owner, "setDefaultConnection", ethDapp.net(), baseConnection.getAddress());
+        xcall.invoke(baseConnection.account, "handleMessage", ethNid, msg.toBytes());
+
+        // Act
+        xcall.invoke(user, "executeCall", BigInteger.ONE, data);
+
+        // Assert
+        verify(defaultDapp.mock).handleCallMessage(ethDapp.toString(), data);
+        verify(xcallSpy).CallExecuted(BigInteger.ONE, 1, "");
+    }
+
```

Comment:
may be we can verify we cant call it again once successful to prevent replay attack.